sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -164,4 +164,11 @@ case class HiveTableScanExec(
   }
 
   override def output: Seq[Attribute] = attributes
+
+  override def sameResult(plan: SparkPlan): Boolean = plan match {
```

Comment:
`left.cleanArgs == right.cleanArgs` in defalut `sameResult` return false, because `equals` in `MetastoreRelation` compare the output(`AttributeReference`) and `exprId`s are diff. We need to erase the exprId.